apps/admin-page/src/app/pages/account/account.component.ts

```diff
@@ -0,0 +1,139 @@
+      (user: IUser) => {
+        this.user = user;
+        this.updateForm();
+        this.isLoading = false;
+      },
+      (err: HttpErrorResponse) => {
+        console.log(err.message);
+        this.isLoading = false;
+      }
+    );
+
+    this.accountForm = new FormGroup({
+      name: new FormControl('', [Validators.maxLength(50), Validators.pattern('^[^0-9]+$')]),
+      phoneNumber: new FormControl('', [
+        Validators.minLength(8),
+        Validators.maxLength(11),
+        Validators.pattern('^[0-9+]*$'),
+      ]),
+      email: new FormControl('', [Validators.required, Validators.email]),
+      streetAddress: new FormControl('', [Validators.maxLength(50), Validators.minLength(7)]),
+      streetAddress2: new FormControl('', [Validators.maxLength(50), Validators.minLength(3)]),
+      city: new FormControl('', [Validators.maxLength(50), Validators.minLength(3), Validators.pattern('^[^0-9]+$')]),
+      postcode: new FormControl('', [Validators.max(9999), Validators.min(555), Validators.pattern('^[0-9]*$')]),
+    });
+  }
+
+  /**
+   * Checks if user is a customer.
+   *
+   * @returns boolean
+   */
+  isCustomer(): boolean | undefined {
+    return !this.user?.roles.includes(UserRoles.developer) || !this.user?.roles.includes(UserRoles.admin);
+  }
+
+  /**
+   * Checks if user is a developer.
+   * @returns boolean
+   */
+  isDeveloper(): boolean | undefined {
+    return this.user?.roles.includes(UserRoles.developer);
+  }
+
+  /**
+   * Checks if a user is an admin.
+   * @returns boolean
+   */
+  isAdmin(): boolean | undefined {
+    return this.user?.roles.includes(UserRoles.admin);
+  }
+
+  /**
+   * Updates the form input fields on the page.
+   */
+  updateForm(): void {
+    this.accountForm.setValue({
+      name: this.user?.name,
+      phoneNumber: this.user?.phoneNumber,
+      email: this.user?.email,
+      streetAddress: this.user?.streetAddress,
+      streetAddress2: this.user?.streetAddress2,
+      city: this.user?.city,
+      postcode: this.user?.postcode,
+    });
+  }
+
+  /**
+   * Updates the account info by sending a request to the api.
+   */
+  updateAccount(): void {
+    this.isUpdatingInfo = true;
+    this.userService
+      .updateUser({
+        name: this.accountForm.get('name')?.value,
+        phoneNumber: this.accountForm.get('phoneNumber')?.value,
+        email: this.accountForm.get('email')?.value,
+        streetAddress: this.accountForm.get('streetAddress')?.value,
+        streetAddress2: this.accountForm.get('streetAddress2')?.value,
+        city: this.accountForm.get('city')?.value,
+        postcode: this.accountForm.get('postcode')?.value,
+      })
+      .subscribe(
+        (data: IUser) => {
+          console.log(data);
+          this.snackBar.open('Your account has been updated!', `I'm the best`, { duration: 5000 });
+          this.isUpdatingInfo = false;
+        },
+        (err: HttpErrorResponse) => {
+          console.log(err.message);
```

Comment:
You should inform the user that the update has failed

Reply:
When logging errors, you should use `console.warn` instead of log for better readability
```suggestion
          console.warn(err.message);
```